core/includes/backup/backup.database.inc

```diff
@@ -0,0 +1,398 @@
+<?php
+/**
+ * @file
+ * Functions to handle the direct to/from database backup source.
+ */
+
+/**
+ * A destination type for saving to a database server.
+ *
+ * @ingroup backup_restore
+ */
+abstract class BackupDatabase extends Backup {
+  /**
+   * @var DatabaseConnection
+   *
+   * The database connection to be used when doing a backup or restore.
+   */
+  protected $connection = NULL;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function typeLabel() {
+    return t('Database');
+  }
+
+  /**
+   * Get the file type supported when backing-up or restoring to/from a file.
+   */
+  public function getFileTypeId() {
+    return 'sql';
+  }
+
+  /**
+   * Get the version info for the given database.
+   */
+  public function databaseInfo() {
+    return array(
+      'type' => FALSE,
+      'version' => t('Unknown'),
+    );
+  }
+
+  /**
+   * Get the default settings for this object.
+   *
+   * @return array
+   *   The default tables whose data can be ignored. These tables mostly
+   *   contain info which can be easily reproduced (such as cache or search
+   *   index) but also tables which can become quite bloated but are not
+   *   necessarily extremely important to back up or migrate during development
+   *   (such as access log and watchdog).
+   */
+  public function defaultSettings() {
+    $settings = parent::defaultSettings();
+
+    $all_tables = $this->getTableNames();
+    $connection = $this->getDatabaseConnection();
+    $prefixes = $connection->getPrefixes();
+
+    // Check if a default database prefix is set. If in use, only back up the
+    // tables within that prefix. If prefixes exist for some tables but there
+    // is no default prefix, maintain the normal behavior of just allowing all
+    // tables.
+    $prefixed_tables = array();
+    $default_prefix = isset($prefixes['default']) ? $prefixes['default'] : NULL;
+    if ($default_prefix) {
+      unset($prefixes['default']);
+      foreach ($all_tables as $table) {
+        foreach ($prefixes as $non_prefixed_table => $prefix) {
+          if ($table == $prefix . $non_prefixed_table) {
+            $prefixed_tables[] = $table;
+          }
+        }
+        if (strpos($table, $default_prefix) === 0) {
+          $prefixed_tables[] = $table;
+        }
+      }
+    }
+
+    // Identify tables that should omit data by default.
+    $nodata_basic = array(
+      // Default core tables.
+      'accesslog',
```

Comment:
There's no such table in core (accesslog).

And I'm not sure, whether we should consider Devel tables here.

Reply:
Yeah I think we can remove the whole list here. :+1: